hdl/jpegenc_v2/myhdl/rgb2ycbcr.py

```diff
@@ -0,0 +1,78 @@
+#!/bin/python
+from myhdl import *
+from random import randrange
+
+Y1, Y2, Y3 = 4899, 9617, 1868
+CB1, CB2, CB3 = -2764, -5428, 8192
+CR1, CR2, CR3 = 8192, -6860, -1332
+Y_OFFSET, CB_OFFSET, CR_OFFSET = 0, 2097152, 2097152
+
+def rgb2ycbcr(y, cb, cr, r, g, b, clk, reset):
+	""" A RGB to YCbCr converter with reset.
+
+	I/O pins:
+	--------
+	y       : output 8-bit unsigned value in range of 0-255
+	cb      : output 8-bit unsigned value in range of 0-255
+	cr      : output 8-bit unsigned value in range of 0-255
+	r 		: input 8-bit unsigned value in range of 0-255
+	g 		: input 8-bit unsigned value in range of 0-255
+	b 		: input 8-bit unsigned value in range of 0-255
+	clk 	: input clock boolean signal
+	reset	: input 8-bit boolean signal
+
+	"""
+
+	@always(clk.negedge, reset.posedge)
+	def logic():
+		if reset:
+			YTemp = intbv(0)[22:]
+			CbTemp = intbv(0)[22:]
+			CrTemp = intbv(0)[22:]
+			y.next = 0
+			cb.next = 0
+			cr.next = 0
+
+		else:
+			YTemp = intbv(Y_OFFSET + Y1 * r + Y2 * g + Y3 * b)[22:]
+			CbTemp = intbv(CB_OFFSET + CB1 * r + CB2 * g + CB3 * b)[22:]
+			CrTemp = intbv(CR_OFFSET + CR1 * r + CR2 * g + CR3 * b)[22:]
+
+			y.next = YTemp[21:14] + 1 if YTemp[13] else YTemp[21:14]
+			cb.next = CbTemp[21:14] + 1 if CbTemp[21:14] != 255 and CbTemp[13] else CbTemp[21:14]
+			cr.next = CrTemp[21:14] + 1 if CrTemp[21:14] != 255 and CrTemp[13] else CrTemp[21:14]
+
+	return logic
+
+def test():
```

Comment:
The test should be in a separate file called test_<mod>.py.  This way py.test or nose can be used as the test-runner, and when this is merged it will be part of the travis-ci.